utils/dataset_voc2coco.py

```
@@ -117,29 +129,33 @@ def get_image_info(annotation_root):
         filename = annotation_root.findtext("filename")
     else:
         filename = os.path.basename(path)
-    img_name = os.path.basename(filename)
+    img_filename = os.path.basename(filename)
+
+    rev_filename = "".join(reversed(img_filename))
+    rev_id_part_filename, _, _ = rev_filename.partition("_")
+    id_part_filename = "".join(reversed(rev_id_part_filename))
+    img_id, img_ext = os.path.splitext(id_part_filename)
 
-    img_id, img_ext = os.path.splitext(img_name)
     if not img_ext in ALLOWED_EXTENSIONS:
-        raise ValueError(f"Image extension is not valid! Got {img_name}")
+        raise ValueError(f"Image extension is not valid! Got {img_filename}")
 
     if not img_id.isdigit():
-        raise ValueError(f"Image file should contain digits only! Got {img_name}")
+        raise ValueError(f"Image file should contain digits only! Got {img_filename}")
 
```

I suggest to add logging here because the error can be ignored when function is invoked

---

We don't need to add logs here because the traceback of the exception will be logged.

---

We don't need to add logs here because the traceback of the exception will be logged.

---

I'm afraid I don't understand you. We intercept this error on the level above and log it, so I guess it is enough.